### ISSUE TITLE
Chakra is slow on a project I'm working on.

### ISSUE THREAD

**Issue by NiLSPACE** [Microsoft/ChakraCore#1262]
With a project I was working on I noticed that Edge is more than 15 times slower than my Firefox.

Project: http://javascript.nilspace.nl/Cuberite/GrownVisualizer/index.html

The pixel rendering (gets activated when the size of the map gets larger than the canvas) is taking ~130ms while it's done in ~8ms on my Firefox, and ~20ms on Chrome. On my mobile Edge is taking over one second.

![edgeversion](https://cloud.githubusercontent.com/assets/1160867/16722837/98caaf88-4749-11e6-9849-a4f1aadb3aeb.PNG)

**Comment by dilijev** [comment]
From profiling it looks like a large part of this time spent is in the DOM:

![image](https://cloud.githubusercontent.com/assets/494055/16748616/f89aebe6-4779-11e6-8d3b-9a9b9dbce859.png)

But plenty of time is used in JS as well.

For the DOM part, could you open a bug at  https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/ and tag it as an HTML/DOM issue?

**Comment by NiLSPACE** [comment]
I created the issue: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/8147766/

I changed the code to put the size of the canvas in a variable
```js
const canvasWidth = canvas.width, canvasHeight = canvas.height;
```
Now the speed is comparable/just as fast as Firefox. That said, while Firefox is roughly constant at 10ms Edge jumps up to 50ms once the map gets to 2000x2000 which then slowly rises with a few milliseconds every time.

I'll wait to update my code on my website so it can be used here for this issue if you'd like.

**Comment by kunalspathak** [comment]
@NiLSPACE , as @dilijev pointed, most of the time is spent in `context.fillRect` in `squareRender()` as seen below. 
![snap1](https://cloud.githubusercontent.com/assets/12488060/16779095/b6769970-4825-11e6-8b72-aafde9d82cf9.PNG)

`pixelRender()` has much time spent in getting the `canvas.height` and `canvas.width` as seen below.
![snap2](https://cloud.githubusercontent.com/assets/12488060/16779140/dbd484ac-4825-11e6-8dce-28eaadc494b7.PNG)

So yes, I agree, caching `canvas.width` and `canvas.height` would give better performance. I would wait for microsoft-edge team to respond on `fillRect()`.

**Comment by NiLSPACE** [comment]
I know the `squareRender` gets really slow as the map gets bigger (that's why I switch to `pixelRender` after a while), but I noticed that `pixelRender` was way slower on Edge compared to Firefox or Chrome. That's why I initially reported this issue :)

As I said in my previous comment, once I changed the code to save the size of the canvas in a variable (as you guys suggested) instead of getting it from the DOM each time the speed is indeed comparable to Firefox (for a while). 

I'm guessing Firefox has the TypedArrays better optimized which keeps the time more constant as the size gets bigger.

The project doesn't need a whole lot performance. In fact, ~130ms is fast enough, but I just thought I should report that Edge is more than 15 times slower than Firefox.
